wrangler-core/src/main/java/io/cdap/wrangler/registry/UserDirectiveRegistry.java

```diff
@@ -90,6 +92,16 @@ public UserDirectiveRegistry(HttpServiceContext manager) throws DirectiveLoadExc
     this.manager = manager;
   }
 
+  /**
+   * This constructor is used when creating from remote task
+   * @param pluginConfigurer {@link PluginConfigurer} for configuring directives
+   * @param artifactManager {@link ArtifactManager} for fetching artifacts for directives
+   */
+  public UserDirectiveRegistry(PluginConfigurer pluginConfigurer, ArtifactManager artifactManager) {
```

Comment:
Adding this constructor will cause a lot of confusion, because either `HttpServiceContext` or `PluginConfiguer` and `ArtifactManager` will be null. I see this constructor is used when using `SystemAppTaskContext`. Our system service uses `SystemHttpServiceContext`, what is the difference between these two contexts, any difference on the plugin configurer and artifact manager they get created?

Reply:
I feel it is better we can have a consistent way of loading artifacts and instantiate plugins

Reply:
SystemAppTaskContext provides only a subset of the contract provided by SystemHttpServiceContext, so cant substitute SystemAppTaskContext in place of SystemHttpServiceContext. But I can use the SystemAppTaskContext directly here instead of assigning PluginConfigurer and ArtifactManager to make it consistent with the existing one.